pkg/build/registry/build/strategy.go

```diff
@@ -110,25 +112,39 @@ type detailsStrategy struct {
 
 // Prepares a build for update by only allowing an update to build details.
 // For now, this is the Spec.Revision field
+// We now need to update the build.status.Reason and build.Status.Message
 func (detailsStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	revision := newBuild.Spec.Revision
+	status := newBuild.Status
 	*newBuild = *oldBuild
 	newBuild.Spec.Revision = revision
+	newBuild.Status = status
 }
 
 // Validates that an update is valid by ensuring that no Revision exists and that it's not getting updated to blank
 func (detailsStrategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	errors := field.ErrorList{}
-	if oldBuild.Spec.Revision != nil {
-		// If there was already a revision, then return an error
-		errors = append(errors, field.Duplicate(field.NewPath("status", "revision"), oldBuild.Spec.Revision))
```

Comment:
@csrwng was this just a mistake that it previously reported the path as "status.revision" instead of "spec.revision"?